contentcuration/contentcuration/frontend/channelList/vuex/channelSet/actions.js

```diff
@@ -55,10 +59,23 @@ export function commitChannelSet(
   for (const channel of channels) {
     channelSetData.channels[channel] = true;
   }
-  return ChannelSet.createModel(channelSetData).then(data => {
-    context.commit('SET_CHANNELSET_NOT_NEW', id);
-    context.commit('UPDATE_CHANNELSET', data);
-  });
+
+  return ChannelSet.createModel(channelSetData)
+    .then(data => {
+      if (!data || !data.id) {
+        console.error('Error: The created channel set does not have an ID', data);
+        throw new Error('The created channel set does not have an ID');
```

Comment:
I think we can throw a less generic error here. We could reuse `ReferenceError`?